### ISSUE TITLE
Doc comment block should strip leading asterisk (*)

### ISSUE THREAD

**Issue by kjeremy** [rust-analyzer/rust-analyzer#1759]
In looking at `struct ClientCapabilities` of lsp-types defined as

```rust
/**
 * Where ClientCapabilities are currently empty:
 */
pub struct ClientCapabilities {
    // ...
}
```

I noticed that hover shows the comment as

> * Where ClientCapabilities are currently empty: 

In looking at https://docs.rs/lsp-types/0.61.0/lsp_types/struct.ClientCapabilities.html we can see  that it's rendered without the leading asterisk (or bullet in markdown).

My initial thought was that maybe we need to strip leading `*` off of block dock comments if present in `DocCommentsOwner::doc_comment_text` but that could also end up stripping markdown. Maybe it should be done in `format_docs` instead when we look for markdown... but by then we've lost all knowledge of the comment except for it's string contents.

**Comment by lnicola** [comment]
Triage: this still happens.

**Comment by patowen** [comment]
This was a bug in Rust nightly itself at some point as well (https://github.com/rust-lang/rust/issues/92872). Since this comment type is shorthand for the `#[doc="..."]` attribute, would it make sense to extract the string from that attribute instead? I assume we don't need any more information about how the comment was written for the help text on mouse hover.

**Comment by Veykril** [comment]
Relevant code is here https://github.com/Veykril/rust-analyzer/blob/9814d798411a4b1b258c710f86626bd1997e406f/crates/hir-def/src/attr.rs#L208-L233

**Comment by boozook** [comment]
> need to strip leading * off of block dock comments if present

This is expected behavior because of the body of (doc)-block-comment is __anything__ between opening and closing tokens, __including any leading or trailing symbols__, according to 
[language spec (grammar)](https://doc.rust-lang.org/reference/grammar.html#railroad-summary-BLOCK_COMMENT) and also body of doc-block-comment is markdown [the rustdoc's doc](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html?highlight=markdown#markdown).

This is not a bug. RA shouldn't strip anything.

**Comment by Veykril** [comment]
As the original post says, rustdoc does not render these leading asterisks, so it is in fact a bug that rust-analyzer doesn't strip these when rendering documentation of multiline doc strings

**Comment by boozook** [comment]
@Veykril are you sure that it's the problem in rust-analyzer, but not bug in rustdoc? I disagree with it. If I want block-doc-comment with only "asterisk-bullet"-list - then it should be so - no one permitted to strip anything, I hope.

**Comment by Veykril** [comment]
Feel free to raise that to the rustdoc folks, we'll be intending to match the behavior with rustdoc, and right now its defined as stripping the initial column of asterisks in these columns

**Comment by boozook** [comment]
> we'll be intending to match the behavior with rustdoc

Do I understand correctly that a specific implementation of rustdoc is more important than the language specification? It’s strong.

**Comment by kivancgnlp** [comment]
Hi! I'd like to pick this up if it's still open and unclaimed. My plan is to strip the leading * column from multiline block doc comments so rendering matches rustdoc, while preserving docs_source_map offsets and adding hover/docs tests. Does that approach still sound right ?

**Comment by ChayimFriedman2** [comment]
@kivancgnlp Yes, you need to change crates/hir-def/src/attrs/docs.rs. You can comment with `@rustbot claim` to assign the issue to you.
